base/applications/regedit/framewnd.c

```diff
@@ -86,6 +88,11 @@ static void OnInitMenu(HWND hWnd)
         while(RemoveMenu(hMenu, s_nFavoriteMenuSubPos, MF_BYPOSITION)) ;
     }
 
+    hSelTI = TreeView_GetSelection(g_pChildWnd->hTreeWnd);
+    bCanAddFav = TreeView_GetParent(g_pChildWnd->hTreeWnd, hSelTI) != NULL;
```

Comment:
or `!!TreeView_GetParent(....)`

Reply:
The existing code is well within 80 columns and `!!` does not make it clearer?

Reply:
it's fine either way.